core/src/main/java/hudson/security/BasicAuthenticationFilter.java

```diff
@@ -140,17 +145,27 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             User u = BasicApiTokenHelper.isConnectingUsingApiToken(username, password);
             if(u != null){
                 UserDetails userDetails = u.getUserDetailsForImpersonation();
-                Authentication auth = u.impersonate(userDetails);
-
-                SecurityListener.fireAuthenticated(userDetails);
-
-                SecurityContextHolder.getContext().setAuthentication(auth);
-                try {
-                    chain.doFilter(request,response);
-                } finally {
-                    SecurityContextHolder.clearContext();
+                if (UserDetailsHelper.isFullyActive(userDetails)) {
+                    Authentication auth = u.impersonate(userDetails);
+
+                    SecurityListener.fireAuthenticated(userDetails);
+
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                    try {
+                        chain.doFilter(request,response);
```

Comment:
space after `,`?

Reply:
1) Even if it's in green, it was out of the scope of the PR as it was "just" indented
2) You can use the GitHub suggestion to ease the modification 

Anyway I corrected it.

Reply:
I'm so not used to that feature, I only just used it for my first time this week… thanks for doing it.